packages/framework/src/Component/Cron/CronTimeResolver.php

```diff
@@ -31,7 +31,7 @@ protected function isMatchWithTimeString($value, $timeString)
         $matches = null;
         foreach ($timeValues as $timeValue) {
             if ($timeValue === '*'
-                || $timeValue === (string)$value
+                || $timeValue === str_pad($value, strlen($timeValue), '0', STR_PAD_LEFT)
```

Comment:
This whole `if` is difficult to understand for me, I would be pleased if you refactored it into 3 functions:
`if ($this->matchesSingleValue($timeValue, $value)`

```php
protected function matchesSingleValue($timeValue, $value) {
return $timeValue === '*' || $this->matchesStringValue($timeValue, $value) || $this->matchesHourPart($timeValue, $value);
}

protected function matchesStringValue($timeValue, $value)...

protected function matchesHourPart($timeValue, $value) ...
```

It would greatly improve readability and maintainability, but it is not "must to be" to pass CR

And of course, you are welcomed to come with better naming

Reply:
I would rather skip the extraction, I don't think it's worth the effort - the result wouldn't be much more readable imo...

Reply:
Just by looking at this line
`preg_match('@^\*/(\d+)$@', $timeValue, $matches) && $value % $matches[1] === 0 // syntax */[int]`

I have no idea what is going on, even with the comment :-D